config/initializers/holidays.rb

```diff
@@ -0,0 +1,51 @@
+Rails.application.config.holidays = [
+  Date.new(2012, 1, 2),
+  Date.new(2012, 4, 6),
+  Date.new(2012, 4, 9),
+  Date.new(2012, 5, 7),
+  Date.new(2012, 6, 4),
+  Date.new(2012, 6, 5),
+  Date.new(2012, 8, 27),
+  Date.new(2012, 12, 25),
+  Date.new(2012, 12, 26),
+  Date.new(2013, 1, 1),
+  Date.new(2013, 3, 29),
+  Date.new(2013, 4, 1),
+  Date.new(2013, 5, 6),
+  Date.new(2013, 5, 27),
+  Date.new(2013, 8, 26),
+  Date.new(2013, 12, 25),
+  Date.new(2013, 12, 26),
+  Date.new(2014, 1, 1),
+  Date.new(2014, 4, 18),
+  Date.new(2014, 4, 21),
+  Date.new(2014, 5, 5),
+  Date.new(2014, 5, 26),
+  Date.new(2014, 8, 25),
+  Date.new(2014, 12, 25),
+  Date.new(2014, 12, 26),
+  Date.new(2015, 1, 1),
+  Date.new(2015, 4, 3),
+  Date.new(2015, 4, 6),
+  Date.new(2015, 5, 4),
+  Date.new(2015, 5, 25),
+  Date.new(2015, 8, 31),
+  Date.new(2015, 12, 25),
+  Date.new(2015, 12, 28),
+  Date.new(2016, 1, 1),
+  Date.new(2016, 3, 25),
+  Date.new(2016, 3, 28),
+  Date.new(2016, 5, 2),
+  Date.new(2016, 5, 30),
+  Date.new(2016, 8, 29),
+  Date.new(2016, 12, 26),
+  Date.new(2016, 12, 27),
+  Date.new(2017, 1, 2),
+  Date.new(2017, 4, 14),
+  Date.new(2017, 4, 17),
+  Date.new(2017, 5, 1),
+  Date.new(2017, 5, 29),
+  Date.new(2017, 8, 28),
+  Date.new(2017, 12, 25),
+  Date.new(2017, 12, 26)
+]
```

Comment:
My proposed workflow is that once a year or so, someone runs the rake task, verifies the changes, and commits them. I don't think it's desirable to automate this. I'll try to explain my reasoning.

1. This data very rarely changes. You'll see that we've got all the public holidays through to the end of 2017, so there's about 24 months before we need to update it.
2. The gov.uk API is not official. Writing something to automate fetching and detecting anomalies seems excessive when it needs updating less than annually: it might change more often than we run it!
3. The API JSON isn't particularly easy to read: it's all rammed together, has information for Scotland and Northern Ireland that is outside the remit of the MOJ, and contains irrelevant cruft like `"bunting": true`. It's much easier to visually check changes in a distilled list that contains only the information we need.
4. We always need this data instantiated, and having it already serialised as Ruby is the simplest way of doing so. It's easy to establish validity, which isn't true of JSON (doesn't even have dates as such) or YAML (too many parse rules). Why have a text-based file and a Ruby file to load them when it's as trivial as this? Equally, we could put it in the database, but that's more complex and less efficient than a linear scan of a couple of dozen dates.

So we could bake something more complicated than this, but I think it would be a waste of time and effort.